Source/JavaScriptCore/API/tests/testapi.c

```diff
@@ -1578,12 +1578,15 @@ int main(int argc, char* argv[])
     configureJSCForTesting();
 
 #if !OS(WINDOWS)
-    char resolvedPath[PATH_MAX];
-    if (!realpath(argv[0], resolvedPath))
+    char *resolvedPath = realpath(argv[0], NULL);
```

Comment:
We don't store memory in raw pointers. Save it directly to a smart pointer:

```suggestion
    std::unique_ptr<char, decltype(free)*> resolvedPath(realpath(argv[0], nullptr), free);
```

Then change uses of `resolvedPath` to `resolvedPath.get()` and remove the call to `free()`.

Reply:
Er, sorry, ignore this suggestion. This is a C source file, so it has to be done manually.